### PR TITLE
Fix bug in name

### PR DESCRIPTION
**Pull request by ajermaky**
Closes # 

## 💸 TL;DR

In #25 I updated library to include additional on behalf of attributes, but in refactoring service checks, i didn't properly propogate it to the `name` property.

Fixing bug + adding additional tests

## 📜 Details
[Design Doc]()

[Jira]()

## ✅ Checks

- [ ] CI tests (if present) are passing
- [ ] Adheres to code style for repo
- [ ] Contributor License Agreement (CLA) completed if not a Reddit employee

### CODE REVIEW

File: lib/py/reddit_edgecontext/__init__.py [hunk 1]
```diff
@@ -436,9 +436,9 @@ def name(self) -> str:
             service.
 
         """
-        subject = self.authentication_token.subject
-        if subject is None or subject == "":
```
Comment (fishy) [inline lib/py/reddit_edgecontext/__init__.py:440]: from my understanding, the bug is that we could just chop the first 8 characters from the t2 id to return here, is that correct?

File: lib/py/tests/edge_context_tests.py [hunk 2]
```diff
@@ -359,3 +365,5 @@ def test_service_auth_with_additional_options(self):
         self.assertEqual(request_context.service.on_behalf_of_id, "t2_deadbeef")
         self.assertEqual(request_context.service.on_behalf_of_roles, ["admin"])
         self.assertEqual(request_context.service.requests_elevated_access, True)
+        with self.assertRaises(NoAuthenticationError):
```
Comment (fishy) [inline lib/py/tests/edge_context_tests.py:368]: why would this case raise `NoAuthenticationError`? this should be a properly service auth'd case?

File: lib/py/reddit_edgecontext/__init__.py (hunk 1 above)
Comment (ajermaky) [reply lib/py/reddit_edgecontext/__init__.py:440]: this is for service auth, so we should be returning a name only if it's prefixed by ('service/')

File: lib/py/tests/edge_context_tests.py (hunk 2 above)
Comment (ajermaky) [reply lib/py/tests/edge_context_tests.py:368]: if it's service auth, then attempting to access user auth should fail (hence `request_context.user.id`)

**Review by fishy — APPROVED** [review]

File: lib/py/tests/edge_context_tests.py [hunk 3]
```diff
@@ -349,6 +353,8 @@ def test_service_auth(self):
         self.assertEqual(request_context.service.on_behalf_of_id, None)
         self.assertEqual(request_context.service.on_behalf_of_roles, None)
         self.assertEqual(request_context.service.requests_elevated_access, None)
+        with self.assertRaises(NoAuthenticationError):
```
Comment (vivekseth) [inline lib/py/tests/edge_context_tests.py:356]: Maybe add a test with a user ID that is longer than 9 characters?

Comment (ajermaky) [reply lib/py/tests/edge_context_tests.py:356]: the issue here was that we were incorrectly characterizing this as service auth - it just so happened that the checks in place would only propogate the issue on longer user ids.

**Pull request merged**